### PR TITLE
240 map list

### PR DESCRIPTION
**Pull request by snixtho**
- New default theme
- Map List (Kinda basic for now)
- Map Queue
- Various fixes
- Manialink-based UI Styles are now separated into their own styles

### New UI Components
- Backdrop
- Chip
- ConfirmDialog
- LinkButton
- Rating
- ScrollBar
- Separator
- Tag
- Circle
- Rectangle

### CODE REVIEW

File: src/EvoSC.Manialinks/Templates/Controls/Text.mt [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  <!-- Y position of the text -->
+  <property type="double" name="y" default="0" />
+  
+  <!-- Whether the text is bold -->
+  <property type="bool" name="bold" default="false" />
+  
+  <template>
+    <label
+            pos="{{ x }} {{ y }}"
+            text='{{ bold ? "$o" : "" }} {{ text }}'
```
Comment (araszka) [inline src/EvoSC.Manialinks/Templates/Controls/Text.mt:23]: Always a space infront of the text

File: src/EvoSC.Manialinks/Templates/Controls/TextInput.mt [hunk 2]
```diff
@@ -2,62 +2,115 @@
...
+                textformat='{{ isPassword ? "Password" : "Basic" }}'
+                name="{{ id }}"
+                data-id="{{ id }}"
+                default="{{ value }}"
+                selecttext="{{ autoSelect }}"
+                maxlen="{{ maxLength }}"
+                z-index="0"
+
+                class="txtinput-text"
+
```
Comment (araszka) [inline src/EvoSC.Manialinks/Templates/Controls/TextInput.mt:62]: Empty lines

File: src/EvoSC.Manialinks/Templates/Drawing/Circle.mt [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+           data-id="{{ dataId }}"
+    >
+        <quad 
+                size="{{ radius*2 }} {{ radius*2 }}"
+                pos="0 0"
+                image="file:///Media/Painter/Stencils/01-EllipseRound/Brush.tga"
+                modulatecolor="{{ bgColor }}"
+                class="{{ className }}"
+                opacity="{{ opacity }}"
+
```
Comment (araszka) [inline src/EvoSC.Manialinks/Templates/Drawing/Circle.mt:55]: Empty line

File: src/EvoSC.Manialinks/Templates/Scripts/Chip.ms [hunk 4]
```diff
@@ -0,0 +1,41 @@
+<!--
+Void CloseChip(Text ChipId) {
+    declare CMlFrame ChipControl <=> (Page.MainFrame.GetFirstChild(ChipId) as CMlFrame);
+    ChipControl.Hide();
+    
+    log("hidden");
```
Comment (araszka) [inline src/EvoSC.Manialinks/Templates/Scripts/Chip.ms:6]: Forgotten debug line?

**Review by araszka — APPROVED** [review]

File: src/Modules/ExampleModule/Templates/DialogExample.mt [hunk 5]
```diff
@@ -0,0 +1,17 @@
+<component>
+  <import component="EvoSC.Controls.ConfirmDialog" as="ConfirmDialog" />
+  <import component="EvoSC.Style.UIStyle" as="UIStyle" />
+  
+  <template>
+    <UIStyle />
+    <ConfirmDialog action="ExampleManialinkController/DialogAction" 
+                   title="you are about to delete a map"
```
Comment (AtomicLiquid) [inline src/Modules/ExampleModule/Templates/DialogExample.mt:8]: Incorrect capitalization

File: src/Modules/MapListModule/Interfaces/IMapListService.cs [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    public Task ShowMapListAsync(IPlayer player);
+    
+    /// <summary>
+    /// Show the confirmation dialog for map deletion to a player. Calling this will
+    /// automatically delete the map if the player confirms.
+    /// </summary>
+    /// <param name="player">Player that needs to confirm the map deletion.</param>
+    /// <param name="map">Map to be deleted.</param>
+    /// <returns></returns>
+    public Task ConfirmMapDeletionsAsync(IPlayer player, IMap map);
```
Comment (AtomicLiquid) [inline src/Modules/MapListModule/Interfaces/IMapListService.cs:38]:
```suggestion
    public Task ConfirmMapDeletionAsync(IPlayer player, IMap map);
```

**Review by AtomicLiquid — APPROVED** [review]
Havent been able to review the tests, but rest looks good, just a few minor things.

File: src/EvoSC.Manialinks/Templates/Scripts/Wigets.ms
[the remark this answers is not in the document]
Comment (snixtho) [reply src/EvoSC.Manialinks/Templates/Scripts/Wigets.ms:1]: typo in the name but intended to be there

**Pull request merged**